src/BenchmarkDotNet/Toolchains/CsProj/CsProjClassicNetToolchain.cs

```diff
@@ -33,21 +36,34 @@ private CsProjClassicNetToolchain(string targetFrameworkMoniker, string name, st
         public static IToolchain From(string targetFrameworkMoniker, string packagesPath = null)
             => new CsProjClassicNetToolchain(targetFrameworkMoniker, targetFrameworkMoniker, packagesPath);
 
-        public override bool IsSupported(BenchmarkCase benchmarkCase, ILogger logger, IResolver resolver)
+        public override IEnumerable<ValidationError> Validate(BenchmarkCase benchmarkCase, ILogger logger, IResolver resolver)
         {
-            if (!base.IsSupported(benchmarkCase, logger, resolver))
-                return false;
+            var baseValidations = base.Validate(benchmarkCase, logger, resolver);
+
+            if (baseValidations.Any())
+            {
+                return baseValidations;
+            }
 
             if (!RuntimeInformation.IsWindows())
             {
-                logger.WriteLineError($"Classic .NET toolchain is supported only for Windows, benchmark '{benchmarkCase.DisplayInfo}' will not be executed");
-                return false;
+                var validationError = new ValidationError(true, $"Classic .NET toolchain is supported only for Windows, benchmark '{benchmarkCase.DisplayInfo}' will not be executed", benchmarkCase);
+
+                logger.WriteLineValidationError(validationError);
+
+                return validationError.AsEnumerable();
             }
 
             if (InvalidCliPath(customDotNetCliPath: null, benchmarkCase, logger))
```

Comment:
You should refactor InvalidCLIPath and return validation errors from it. After it you will can delete logger